CHANGELOG.md

```diff
@@ -1,5 +1,39 @@
+* Support for all types of attribute values, not just strings:
+    * All values are passed through to notification listeners.
+    * Strings, booleans, and valid numbers are passed to the event dispatcher and can be used for Optimizely results segmentation. A valid number is a finite number in the inclusive range [-2⁵³, 2⁵³]. 
+    * Strings, booleans, and valid numbers are relevant for audience conditions.
+* Support for additional matchers in audience conditions:
+    * An `exists` matcher that passes if the user has a non-null value for the targeted user attribute and fails otherwise.
+    * A `substring` matcher that resolves if the user has a string value for the targeted attribute.
+    * `gt` (greater than) and `lt` (less than) matchers that resolve if the user has a valid number value for the targeted attribute. A valid number is a finite number in the inclusive range [-2⁵³, 2⁵³].
+    * The original (`exact`) matcher can now be used to target booleans and valid numbers, not just strings.
+* Support for A/B tests, feature tests, and feature rollouts whose audiences are combined using `"and"` and `"not"` operators, not just the `"or"` operator.
+* Datafile-version compatibility check: The SDK will remain uninitialized (i.e., will gracefully fail to activate experiments and features) if given a datafile version greater than 4.
+* Updated Pull Request template and commit message guidelines.
+* When given an invalid datafile, the Optimizely client object now instantiates into a no-op state instead of throwing a `ConfigParseException`. This matches the behavior of the other Optimizely SDKs.
+* Support for graceful shutdown in the default, async event dispatcher.
+### Breaking Changes:
+* Java 7 is no longer supported.
+* Previously, notification listeners were only given string-valued user attributes because only strings could be passed into various method calls. That is no longer the case. The `ActivateNotificationListener` and `TrackNotificationListener` interfaces now receive user attributes as `Map<String, ?>` instead of `Map<String, String>`.
+### Bug Fixes:
+* Experiments and features can no longer activate when a negatively targeted attribute has a missing, null, or malformed value.
+* Audience conditions (except for the new `exists` matcher) no longer resolve to `false` when they fail to find an legitimate value for the targeted user attribute. The result remains `null` (unknown). Therefore, an audience that negates such a condition (using the `"not"` operator) can no longer resolve to `true` unless there is an unrelated branch in the condition tree that itself resolves to `true`.
+* Support for empty user IDs.
+* Wrap the buffer reader in try...catch.
```

Comment:
Should we have PRs linked here? @mikeng13 ?

Reply:
Yes, these should like to the respective PRs like we do in previous release notes.

Reply:
Hmm, we seemed to be inconsistent on that front... I did omit in the centralized draft of the release notes, primarily for logistical reasons, but also because some changes couldn't be described by a single PR.

I'd say SDK releasers are free to add links, but they're not easy to determine for _all_ changes, and I personally am not gonna require them.

Reply:
We've been inconsistent before but we should be consistent now. Going forward we should do it all our releases!